compal/__init__.py

```diff
@@ -672,30 +696,206 @@ def transform_radio(radio_settings):  # rs = radio_settings
+        """
+
+        def iterate_changes(old_settings, new_settings, changes):
+            for attr, old_value in old_settings.__dict__.items():
+                if isinstance(old_value, BandSetting):
+                    changes.update({attr: {}})
+                    iterate_changes(getattr(old_settings, attr), getattr(new_settings, attr), changes[attr])
+                else:
+                    new_value = getattr(new_settings, attr)
+                    if old_value != new_value:
+                        changes.update({attr: f"{old_value} -> {new_value}"})
+                        bln_changes[0] = True
+
+        changes = {}
+        bln_changes = [False]
+        iterate_changes(old_settings, new_settings, changes)
+        return bln_changes[0], changes
+
+    def __check_router_status(self, new_settings, debug=True):
+        """
+        Checks if (1) the new user settings were changed in the router (via checking
+        the getter fun:300 [WIRELESSBASIC] in the router-xml-file), (2) prints out a
+        progress bar and (3) prints out if the process was successful.
+        """
+        router_settings = None
+
+        def print_progress_bar(progress, total, postfix=""):
+            progress = total if progress > total else progress
+            per_progress = int(progress / total * 100)
+            postfix = f"[{postfix}]" if postfix != "" else ""
+            print(f"\r|{'█' * progress}{'-' * (total - progress)}| {per_progress}%, {progress}sec\t{postfix}", end="")
+
+        progress = 0
+        total = 24
+        bln_changes = True
+        start_time = time.time()
+        changes = ""
+        if debug:
+            print(f"\n--- WAITING FOR ROUTER TO UPDATE ---")
+        while progress < total and bln_changes is True:
+            progress = int(time.time() - start_time)
+            if debug:
+                print_progress_bar(progress, total, changes)
+            time.sleep(3)
+            try:
+                router_settings = self.wifi_settings
+                bln_changes, changes = self.__compare_wifi_settings(router_settings, new_settings)
+            except Exception as e:
+                changes = str(e)
+        if debug:
+            if not bln_changes:
+                print(f"\n\n--- ROUTER SUCESSFULLY UPDATED ALL NEW WIFI SETTINGS! ---")
+            else:
+                print(f"\n\n--- CHANGES THAT DID NOT GET SET ---")
+                _, changes = self.__compare_wifi_settings(router_settings, new_settings)
+                print(changes)
+        return bln_changes
+
+    @staticmethod
+    def __update_new_settings(old_settings, new_settings):
+        """
+        Needed for 'self.update_wifi_settings', if the user only changes
+        band_mode or only changes radio_2g.bss_enable or radio_5g.bss_enable.
+        """
+        new = new_settings
+        if old_settings.band_mode != new.band_mode:
+            new.radio_2g.bss_enable = 1 if (new_settings.band_mode & 1) else 2
```

Comment:
I think `new_settings is new` (the objects are the same). So the method is still updating `new_settings`. Not sure if that is what you intended :)

Reply:
Humm... i think i didn't found a better name for the function. The problem is always when the user only changes the bandmode or the radio_2g.bss_enable/radio_5g.bss_enable. After knowing, which of the variables the user changed (bandmode or bss_enable) the new_settings file has to update. If not, the comparing of the dictionaries fails and it starts a new api request to fun:301/fun:319. 
I think you see best what this function does, if you run update_wifi_settings without it (ps: this time i click "start a review" just to test what it makes).

Reply:
It was just a style nit. Not important. I personally stay away from the word `new` for variables because it is a keyword in many programming languages.

What the code does is clear and that is the important part 😊